### DISCUSSION TITLE
this-as or calling globaljs/objects from a cljs function

### DISCUSSION THREAD

**Discussion by frap** [babashka/nbb#218] [Q&A] [answered]
I'm not the best on javascript programming however using ssh2 I can get it to work in the "global" the  namespace of the file aka ssh.cljs. However when I try to put the code in a function I keep getting 
` Je: Error: No namespace: null found` errors. 
 then tried to use the this-as function and found its not implemented. Is my reasoning correct its because the imported ssh2 Client can't be called from an cljs function?

non working code (that works if you put in "global" namespace of file but just not in the ssh-exec fn
`(ns ssh
   (require   ["ssh2" :refer [Client]]
            ))
(defn ssh-exec [command]
  (let [client       (Client.)
        conn         (.connect client sshconfig)]
    (-> conn
        (.on "error" (fn [err] (println (str "ERROR:: " err))))
        (.on "ready"
             (fn [] (println "stream ready")))
        (.exec conn command
               (fn [err stream]
                 ;;(if err (throw err))
                 (.on stream "close"
                      (fn [code signal] (println (str "STRM:: close ::  " code))
                        (.end conn)))
                 (.on stream "data"
                      (fn [data] (println (str "STDOUT:: " data))))
                 )))))`

The code doesnt work if I pass in the Client constructor into the function either

Sorry if this is a *noob* question

**Comment by borkdude** [comment, marked as answer]
@frap I don't know how this-as relates to your question, but I made an example with ssh2 here now:

https://github.com/babashka/nbb/blob/main/examples/ssh2/example.cljs

It also shows a promise-based approach using promise and ssh2-promise which might be more convenient.

**Reply by borkdude** [reply]
`this-as` isn't supported in nbb btw. Instead there are workarounds for it, but I don't see yet how it relates to this example.

**Reply by frap** [reply]
Thanks @borkdude. Looking at your code my issue was converting #js method point calls to clojurescript functions and the -> threading macro. the null namespace errors in nbb were detouring 
me into javascript this concept where my actual issue was more basic :(
